HeinOnline.js

```diff
@@ -0,0 +1,158 @@
+{
+	"translatorID": "3dcbb947-f7e3-4bbd-a4e5-717f3701d624",
+	"label": "HeinOnline",
+	"creator": "Frank Bennett",
+	"target": "^https?://heinonline\.org/HOL/(?:LuceneSearch|Page)\\?",
+	"minVersion": "1.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "g",
+	"lastUpdated": "2015-07-04 10:32:10"
+}
+
+function detectWeb (doc, url) {
+	if (url.indexOf("/LuceneSearch?") > -1) {
+		return "multiple";
+	} else {
+		return "journalArticle";
+	}
+}
+
+function Data(doc) {
+	this.node = ZU.xpath(doc, '//form[@id="Print1"]');
+	var urlLst = doc.location.href.split('/');
+	this.urlbase = urlLst.slice(0, 4).join('/') + "/PDFsearchable?";
+	this.queryElems = [];
+	this.tail = "&sectioncount=1&ext=.pdf&nocover=";
+}
+
+Data.prototype.getval = function(name, returnOnly) {
+	var val = '';
+	var input = ZU.xpath(this.node, './/input[@name="' + name + '"]');
+	if (input && input.length) {
+		val = input[0].value ? encodeURIComponent(input[0].value) : '';
+	}
+	if (!returnOnly) {
+		this.queryElems.push(name + "=" + val);
```

Comment:
I'm sure it will be ok as is.
On Jul 4, 2015 4:17 PM, "Frank Bennett" <notifications@github.com> wrote:

> In HeinOnline.js
> <https://github.com/zotero/translators/pull/915#discussion_r33890079>:
>
> > +function Data(doc) {
> > +	this.node = ZU.xpath(doc, '//form[@id="Print1"]');
> > +	var urlLst = doc.location.href.split('/');
> > +	this.urlbase = urlLst.slice(0, 4).join('/') + "/PDFsearchable?";
> > +	this.queryElems = [];
> > +	this.tail = "&sectioncount=1&ext=.pdf&nocover=";
> > +}
> > +
> > +Data.prototype.getval = function(name, returnOnly) {
> > +	var val = '';
> > +	var input = ZU.xpath(this.node, './/input[@name="' + name + '"]');
> > +	if (input && input.length) {
> > +		val = input[0].value ? encodeURIComponent(input[0].value) : '';
> > +	}
> > +	if (!returnOnly) {
> > +		this.queryElems.push(name + "=" + val);
>
> Okay to leave this as is, since the values of "name" are known to not to
> be ones that require encoding? Or would you prefer to have a wrapper on it
> for completeness?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zotero/translators/pull/915/files#r33890079>.
>